sdk/storage/azure-storage-blob/V10_V12.md

```diff
@@ -0,0 +1,379 @@
+```java
+// specify token properties
+ServiceSASSignatureValues value = new ServiceSASSignatureValues();
+BlobSASPermission p = new BlobSASPermission()
+    .withRead(true)
+    .withWrite(true)
+    .withCreate(true)
+    .withDelete(true)
+    .withAdd(true);
+value.withPermissions(p.toString())
+    .withStartTime(OffsetDateTime.now().minusDays(1))
+    .withExpiryTime(OffsetDateTime.now().plusDays(1))
+    .withContainerName("<container-name>")
+    .withBlobName("<blob-name>");
+IPRange ipR = new IPRange()
+    .withIpMin("0.0.0.0")
+    .withIpMax("255.255.255.255");
+value.withIpRange(ipR)
+    .withProtocol(SASProtocol.HTTPS_ONLY)
+    .withCacheControl("cache")
+    .withContentDisposition("disposition")
+    .withContentEncoding("encoding")
+    .withContentLanguage("language")
+    .withContentType("type");
+
+// build the token
+SASQueryParameters sasToken = value.generateSASQueryParameters(sharedKeyCredentails);
+```
+
+In V12, SAS tokens are generated off the client to the resource you wish to generate the SAS for.
+The resource path handled automatically because of this, and the `SharedKeyCredential` authenticating the client is also used automatically to sign the SAS. See [building a client](#buildclient) for how to get an authenticated client.
+```java
+// specify token properties
+String identifier = "identifier";
+BlobSASPermission permissions = new BlobSASPermission()
+    .withRead(true)
+    .withCreate(true)
+    .withDelete(true)
+    .withWrite(true); // We can also choose BlobContainerSasPermission, BlobServiceSasQueryParameters based on the object level we want to grant access.
+OffsetDateTime startTime = OffsetDateTime.now().minusDays(1);
+OffsetDateTime expiryTime = OffsetDateTime.now().plusDays(1);
+SasIpRange ipRange = new SasIpRange()
+    .setIpMax("0.0.0.0")
+    .setIpMax("255.255.255.255");
+SasProtocol sasProtocol = SasProtocol.HTTPS_HTTP;
+String cacheControl = "cache";
+String contentDisposition = "disposition";
+String contentEncoding = "encoding";
+String contentLanguage = "language";
+String contentType = "type";
+String version = BlobServiceVersion.V2019_02_02.getVersion();
+
+// build the token
+BlobServiceSasSignatureValues sasSignatureValues = new BlobServiceSasSignatureValues(version, sasProtocol, startTime, expiryTime, permissions.toString(), ipRange, identifier, cacheControl, contentDisposition, contentEncoding, contentLanguage, contentType);
+BlobServiceSasQueryParameters sasQueryParameters = sasSignatureValues.generateSasQueryParameters(new StorageSharedKeyCredential("<account-name>", "<account-key"));
+```
+
+## Build Client
+In V10/V11, core classes were built with constructors, taking a `URL` and an already constructed `HttpPipeline`.
```

Comment:
I think this section should go before sas tokens because you need to build a client in order to generate a sas token

Reply:
Done.

Reply:
Done.